### PR TITLE
Drop Convert static constructor dependency from RuntimeType

### PR DESCRIPTION
**Pull request by marek-safar**
to make it trimmable. I also cleaned up GetTypeCode calls chain along the way.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs [hunk 1]
```diff
@@ -194,23 +194,34 @@ protected override TypeCode GetTypeCodeImpl()
                     typeCode = TypeCode.Single; break;
                 case CorElementType.ELEMENT_TYPE_R8:
                     typeCode = TypeCode.Double; break;
+                case CorElementType.ELEMENT_TYPE_STRING:
+                    typeCode = TypeCode.String; break;
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs:198]:
```suggestion
#if !CORECLR
                case CorElementType.ELEMENT_TYPE_STRING:
                    typeCode = TypeCode.String; break;
#endif
```

**Comment by jkotas** [comment]
Does it have measurable impact on GetTypeCode() performance? (It should not, but it may be worth it to double check.)

**Comment by marek-safar** [comment]
The whole switch is ran only once per type https://github.com/dotnet/runtime/blob/dc019d7f2e680712e6538766f3fac5af027aeac4/src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs#L167.  I'm sceptical this change will be measurable.

**Comment by jkotas** [comment]
Ah ok. I have not realized that the result is cached. LGTM.

**Review by jkotas — APPROVED** [review]

**Pull request merged**